src/Admin/fields/AssociatedDonor/index.tsx

```diff
@@ -0,0 +1,52 @@
+import AdminSection, {AdminSectionField} from '@givewp/components/AdminDetailsPage/AdminSection';
+import {__} from '@wordpress/i18n';
+import {useFormContext, useFormState} from 'react-hook-form';
+import {AsyncPaginate} from 'react-select-async-paginate';
+import styles from './styles.module.scss';
+import {DonorOption} from './types';
+import {useDonorAsyncSelect} from './useDonorAsyncSelect';
+
+/**
+ * @unreleased add mode prop
+ * @since 4.9.0 Add error prop to all AdminSectionField components
+ * @since 4.8.0 updated to async donor dropdown
+ * @since 4.6.0
+ */
+export default function AssociatedDonor({mode}: {mode: 'test' | 'live'}) {
+    const {watch, setValue} = useFormContext();
+    const {errors} = useFormState();
+    const donationDonorId = watch('donorId');
```

Comment:
As a shared field, we should make this more agnostic without making too many assumptions.  For example, I think we should add a prop for `name` instead of assuming the field name will be `donorId` as that might not always be the case inside a form.  We should also just use the word `donorId` instead of `donationDonorId` as that implies context.

```suggestion
    const donorId = watch(name);
```

Reply:
Resolved: https://github.com/impress-org/givewp/pull/8119/commits/78fbb4bb917f14cc3b7e2280f1acdc6b76181411